diff -rupN 4.8.5-src/mkspecs/qws/vnc-linux-x86_64-g++/qmake.conf 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86_64-g++/qmake.conf
--- 4.8.5-src/mkspecs/qws/vnc-linux-x86_64-g++/qmake.conf	1969-12-31 18:00:00.000000000 -0600
+++ 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86_64-g++/qmake.conf	2014-02-22 19:29:56.747671358 -0600
@@ -0,0 +1,19 @@
+#
+# qmake configuration for building with linux-g++
+#
+
+include(../../common/linux.conf)
+include(../../common/gcc-base-unix.conf)
+include(../../common/g++-unix.conf)
+include(../../common/qws.conf)
+
+# modifications to g++.conf
+QMAKE_CFLAGS            += -m64
+QMAKE_CXXFLAGS          += -m64
+
+CONFIG                  += vnc
+
+DEFINES                 += VNCSERVER
+
+load(qt_config)
+
diff -rupN 4.8.5-src/mkspecs/qws/vnc-linux-x86_64-g++/qplatformdefs.h 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86_64-g++/qplatformdefs.h
--- 4.8.5-src/mkspecs/qws/vnc-linux-x86_64-g++/qplatformdefs.h	1969-12-31 18:00:00.000000000 -0600
+++ 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86_64-g++/qplatformdefs.h	2014-02-22 19:29:56.735671358 -0600
@@ -0,0 +1,42 @@
+/****************************************************************************
+**
+** Copyright (C) 2013 Digia Plc and/or its subsidiary(-ies).
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia.  For licensing terms and
+** conditions see http://qt.digia.com/licensing.  For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights.  These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 3.0 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU General Public License version 3.0 requirements will be
+** met: http://www.gnu.org/copyleft/gpl.html.
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../../linux-g++/qplatformdefs.h"
diff -rupN 4.8.5-src/mkspecs/qws/vnc-linux-x86-g++/qmake.conf 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86-g++/qmake.conf
--- 4.8.5-src/mkspecs/qws/vnc-linux-x86-g++/qmake.conf	1969-12-31 18:00:00.000000000 -0600
+++ 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86-g++/qmake.conf	2014-02-22 19:29:56.771671357 -0600
@@ -0,0 +1,18 @@
+#
+# qmake configuration for building with linux-g++
+#
+
+include(../../common/linux.conf)
+include(../../common/gcc-base-unix.conf)
+include(../../common/g++-unix.conf)
+include(../../common/qws.conf)
+
+# modifications to g++.conf
+QMAKE_CFLAGS            += -m32
+QMAKE_CXXFLAGS          += -m32
+
+CONFIG                  += vnc       
+
+DEFINES                 += VNCSERVER
+
+load(qt_config)
diff -rupN 4.8.5-src/mkspecs/qws/vnc-linux-x86-g++/qplatformdefs.h 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86-g++/qplatformdefs.h
--- 4.8.5-src/mkspecs/qws/vnc-linux-x86-g++/qplatformdefs.h	1969-12-31 18:00:00.000000000 -0600
+++ 4.8.5-src-patched-vnc/mkspecs/qws/vnc-linux-x86-g++/qplatformdefs.h	2014-02-22 19:29:56.759671358 -0600
@@ -0,0 +1,42 @@
+/****************************************************************************
+**
+** Copyright (C) 2013 Digia Plc and/or its subsidiary(-ies).
+** Contact: http://www.qt-project.org/legal
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** Commercial License Usage
+** Licensees holding valid commercial Qt licenses may use this file in
+** accordance with the commercial license agreement provided with the
+** Software or, alternatively, in accordance with the terms contained in
+** a written agreement between you and Digia.  For licensing terms and
+** conditions see http://qt.digia.com/licensing.  For further information
+** use the contact form at http://qt.digia.com/contact-us.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Digia gives you certain additional
+** rights.  These rights are described in the Digia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU
+** General Public License version 3.0 as published by the Free Software
+** Foundation and appearing in the file LICENSE.GPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU General Public License version 3.0 requirements will be
+** met: http://www.gnu.org/copyleft/gpl.html.
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../../linux-g++/qplatformdefs.h"
diff -rupN 4.8.5-src/src/gui/kernel/qclipboard_qws.cpp 4.8.5-src-patched-vnc/src/gui/kernel/qclipboard_qws.cpp
--- 4.8.5-src/src/gui/kernel/qclipboard_qws.cpp	2013-06-07 00:16:59.000000000 -0500
+++ 4.8.5-src-patched-vnc/src/gui/kernel/qclipboard_qws.cpp	2014-02-26 21:16:38.000000000 -0600
@@ -53,6 +53,8 @@
 #include <qwsdisplay_qws.h>
 #include <qwsproperty_qws.h>
 #include <qwsevent_qws.h>
+#include <QScreen>
+#include <qscreenvnc_qws.h>
 
 QT_BEGIN_NAMESPACE
 
@@ -100,6 +102,10 @@ static void qwsSetClipboardText(const QS
     QPaintDevice::qwsDisplay()->
         setProperty(0, TextClipboard, QWSPropertyManager::PropReplace, ba);
 
+    QScreen *pScreen = QScreen::instance();
+    if ((pScreen) && (pScreen->classId() == QScreen::VNCClass)) {
+        (static_cast<QVNCScreen *>(pScreen))->serverCutText();
+    }
 }
 
 class QClipboardData
diff -rupN 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_p.h 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_p.h
--- 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_p.h	2013-06-07 00:17:00.000000000 -0500
+++ 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_p.h	2014-02-26 21:29:38.000000000 -0600
@@ -450,7 +450,8 @@ public:
                      ClientCutText = 6 };
 
     enum ServerMsg { FramebufferUpdate = 0,
-                     SetColourMapEntries = 1 };
+                     SetColourMapEntries = 1,
+                     ServerCutText = 3 };
 
     void convertPixels(char *dst, const char *src, int count) const;
 
@@ -467,6 +468,8 @@ public:
     inline bool hasClientCursor() const { return qvnc_cursor != 0; }
 #endif
 
+    void serverCutText();
+
 private:
     void setPixelFormat();
     void setEncodings();
diff -rupN 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.cpp 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.cpp
--- 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.cpp	2013-06-07 00:17:00.000000000 -0500
+++ 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.cpp	2014-02-26 22:48:04.000000000 -0600
@@ -60,6 +60,10 @@
 
 #include <stdlib.h>
 
+#include <QApplication>
+#include <QClipboard>
+#include <QByteArray>
+
 QT_BEGIN_NAMESPACE
 
 //#define QT_QWS_VNC_DEBUG
@@ -1062,12 +1066,33 @@ void QVNCServer::clientCutText()
     if (cutTextPending && client->bytesAvailable() >= cutTextPending) {
         char *text = new char [cutTextPending+1];
         client->read(text, cutTextPending);
+        text[cutTextPending] = 0;		// Make sure string is terminated
+        // Ideally, we'd use UTF8, but according to the RFB spec, the clipboard
+        //	operations are limited to Latin1:
+        QApplication::clipboard()->setText(QString::fromLatin1(text, cutTextPending));
         delete [] text;
         cutTextPending = 0;
         handleMsg = false;
     }
 }
 
+void QVNCServer::serverCutText()
+{
+    QTcpSocket *socket = clientSocket();
+
+    // According to the RFB spec, the clipboard only supports Latin1 encoding:
+    QByteArray baClipboard = QApplication::clipboard()->text().toLatin1();
+
+    // ServerCutText header
+    const char cmd = 3;
+    socket->write(&cmd, sizeof(cmd));
+    const char padding[3] = { 0, 0, 0 };
+    socket->write(padding, sizeof(padding));
+    const quint32 nSize = htonl(baClipboard.size());
+    socket->write(reinterpret_cast<const char *>(&nSize), sizeof(nSize));
+    socket->write(baClipboard.data(), baClipboard.size());
+}
+
 // stride in bytes
 template <class SRC>
 bool QRfbSingleColorHextile<SRC>::read(const uchar *data,
@@ -2333,6 +2358,11 @@ void QVNCScreen::shutdownDevice()
     delete d_ptr->dirty;
 }
 
+void QVNCScreen::serverCutText()
+{
+    if (d_ptr->vncServer) d_ptr->vncServer->serverCutText();
+}
+
 QT_END_NAMESPACE
 
 #endif // QT_NO_QWS_VNC
diff -rupN 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.h 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.h
--- 4.8.5-src/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.h	2013-06-07 00:17:00.000000000 -0500
+++ 4.8.5-src-patched-vnc/src/plugins/gfxdrivers/vnc/qscreenvnc_qws.h	2014-02-26 20:38:42.000000000 -0600
@@ -67,6 +67,8 @@ public:
 
     void setDirty(const QRect&);
 
+    void serverCutText();
+
 private:
     friend class QVNCCursor;
     friend class QVNCClientCursor;
